roles/deploy/tasks/update.yml

```diff
@@ -6,7 +6,7 @@
   known_hosts:
     name: "{{ item.name }}"
     key: "{{ item.key | default(omit) }}"
-    path: "{{ item.path | default('/home/' + ansible_user + '/.ssh/known_hosts') }}"
+    path: "{{ item.path | default(home_root + ansible_user + '/.ssh/known_hosts') }}"
```

Comment:
How about using [`default(omit)`](http://docs.ansible.com/ansible/playbooks_filters.html#omitting-parameters) instead?
`path: "{{ item.path | default(omit) }}"`

It should default to `~/.ssh/known_hosts` ([docs](http://docs.ansible.com/ansible/known_hosts_module.html) and [src](https://github.com/ansible/ansible-modules-extras/blob/stable-2.2/system/known_hosts.py#L274)). This works in my tests (but I didn't test custom home dir). I probably should have used `omit` in #751.

If this works for your user's custom home directory, we could use `omit` for both tasks' `path` and no longer need a new `home_root` variable.